### PR TITLE
LCAM-53

### PR DESCRIPTION
**Pull request by Mhowell494**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-53)

- New update endpoint
- Refactored controllers, services and builders
- Removed redundant JMS configuration code
- Slimmed down schemas by removing redundant fields
- New validator class that checks the full assessment date.

### CODE REVIEW

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/validation/service/MeansAssessmentValidationService.java [hunk 1]
```diff
@@ -32,11 +32,15 @@ public class MeansAssessmentValidationService {
     private final WebClient webClient;
     private final MaatApiConfiguration configuration;
 
+    String getUserIdFromRequest(MeansAssessmentRequestDTO meansAssessmentRequest) {
+        return meansAssessmentRequest.getUserSession().getUserName();
```
Comment (venkat980) [inline crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/validation/service/MeansAssessmentValidationService.java:36]: Same comment as above, should we be checking if the UserSession is not null here?

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/builder/MaatCourtDataAssessmentBuilder.java [hunk 2]
```diff
@@ -75,7 +73,7 @@ private MaatApiCreateAssessment buildCreate(MeansAssessmentRequestDTO requestDTO
                 .withRtCode(ofNullable(requestDTO.getReviewType())
                         .map(ReviewType::getCode).orElse(null))
                 .withNworCode(requestDTO.getNewWorkReason().getCode())
-                .withUserCreated(requestDTO.getUserId())
+                .withUserCreated(requestDTO.getUserSession().getUserName())
```
[the remark this answers is not in the document]
Comment (Mhowell494) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/builder/MaatCourtDataAssessmentBuilder.java:76]: The userSession and the values it contains are all required fields in the schema, so any request that was missing those fields would be rejected with a 400.

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/validation/service/MeansAssessmentValidationService.java (hunk 1 above)
Comment (Mhowell494) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/validation/service/MeansAssessmentValidationService.java:36]: The userSession and the values it contains are all required fields in the schema, so any request that was missing those fields would be rejected with a 400.

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/aspect/RequestLoggingHandler.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @Around("execution(public * uk.gov.justice.laa.crime.meansassessment.controller.MeansAssessmentController.*(..))")
+    public Object logAroundControllers(ProceedingJoinPoint pjp) throws Throwable {
+        Object result;
+        String methodName = pjp.getSignature().getName();
+        ApiMeansAssessmentRequest meansAssessment = (ApiMeansAssessmentRequest) pjp.getArgs()[0];
+        MDC.put(LoggingData.MAAT_ID.getValue(), Integer.toString(meansAssessment.getRepId()));
+        log.info("{} request received for MAAT ID: {}", methodName, MDC.get(LoggingData.MAAT_ID.getValue()));
+        result = pjp.proceed();
+        log.info("{} request completed for MAAT ID: {}", methodName, MDC.get(LoggingData.MAAT_ID.getValue()));
+        MDC.remove(LoggingData.MAAT_ID.getValue());
```
[the remark this answers is not in the document]
Comment (Mhowell494) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/aspect/RequestLoggingHandler.java]: Good suggestions, I've fixed this in the latest commit :)

**Review by venkat980 — APPROVED** [review]

**Pull request merged**